_P221_LandisGyrUH550.ino

```diff
@@ -0,0 +1,450 @@
+  byte vupCount;
+  struct ValueUnitPair vup[20];
+};
+
+struct DataSet datasets[10];
+byte indexDS = 0;
+bool messageProcessingComplete = false;
+
+boolean Plugin_221(byte function, struct EventStruct *event, String &string)
+{
+  boolean success = false;
+
+  Serial.begin(115200);
+
+  switch (function)
+  {
+    case PLUGIN_DEVICE_ADD:
+    {
+      Device[++deviceCount].Number = PLUGIN_ID_221;
+      Device[deviceCount].Type = DEVICE_TYPE_DUAL;
+      Device[deviceCount].VType = SENSOR_TYPE_SINGLE;
+      Device[deviceCount].Ports = 0;
+      Device[deviceCount].PullUpOption = false;
+      Device[deviceCount].InverseLogicOption = false;
+      Device[deviceCount].FormulaOption = true;
+      Device[deviceCount].ValueCount = 1;
+      Device[deviceCount].SendDataOption = true;
+      Device[deviceCount].TimerOption = true;
+      Device[deviceCount].GlobalSyncOption = true;
+      break;
+    }
+
+    case PLUGIN_GET_DEVICENAME:
+    {
+      string = F(PLUGIN_NAME_221);
+      break;
+    }
+
+    case PLUGIN_GET_DEVICEVALUENAMES:
+    {
+      strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[0], PSTR(PLUGIN_VALUENAME1_221));
+      break;
+    }
+
+    case PLUGIN_INIT:
+    {
+      Plugin_221_init = true;
+
+      success = true;
+      break;
+    }
+
+    case PLUGIN_WEBFORM_LOAD:
+    {
+      byte valueSet = Settings.TaskDevicePluginConfig[event->TaskIndex][1];
+
+      String optionsValueSet[7] = {  F("Quantity of Heat"),
+                                      F("Volume"),
+                                      F("Quantity of Heat previous Year"),
+                                      F("Volume previous Year"),
+                                      F("maxPower"),
+                                      F("maxInflowTemp"),
+                                      F("maxReflowTemp") };
+      addFormSelector(F("Valueset"), F("plugin_221_query"), 7, optionsValueSet, NULL, valueSet );
+
+      success = true;
+      break;
+    }
+
+    case PLUGIN_WEBFORM_SAVE:
+    {
+      Settings.TaskDevicePluginConfig[event->TaskIndex][0] = getFormItemInt(F("plugin_221"));
+      Settings.TaskDevicePluginConfig[event->TaskIndex][1] = getFormItemInt(F("plugin_221_query"));
+
+      Plugin_221_init = false; // Force device setup next time
+      success = true;
+      break;
+    }
+
+    case PLUGIN_READ:
+    {
+
+      if (Plugin_221_init)
+      {
+        String log = F("LUGUH: start read cycle: task <nr> : ");
+        log += event->TaskIndex;
+        log += " : "; 
+        log += getTimeString(':', true);
+        addLog(LOG_LEVEL_INFO, log);
+
+        switch(Settings.TaskDevicePluginConfig[event->TaskIndex][1])
+        {
+          case 0:
+          {
+            messageProcessingComplete = false;
+
+            PIN_SER_RX = Settings.TaskDevicePin1[event->TaskIndex];
+            PIN_SER_TX = Settings.TaskDevicePin2[event->TaskIndex];
+
+            ESPeasySoftwareSerial softSerial(PIN_SER_RX, PIN_SER_TX, false); // Initialize serial

```

Comment:
This should use the `ESPeasySerial` wrapper, to make it transparent for using the HWserial or SWserial.
See for some examples the GPS plugin (Also the helper class in the load and save part of the code)